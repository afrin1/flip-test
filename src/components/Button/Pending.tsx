import React from "react";
import { Alert, StyleSheet, Text, TouchableOpacity } from "react-native";
import { ButtonProps } from ".";
import { colors } from "../../theme/colors";
import { buttonText } from "../../theme/typography";

const PendingButton = (props: ButtonProps) => {
  const { title, onPress } = props
  return (
    <TouchableOpacity
      onPress={onPress}
      style={styles.button}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    display: 'flex',
    flexDirection: 'row',
    padding: 10,
    backgroundColor: '#FFFFFF',
    borderRadius: 10,
    borderColor: colors.PENDING,
    borderWidth: 2,
  },
  buttonText: {
    color: '#000000',
    ...buttonText as any
  },
})

export default PendingButton;