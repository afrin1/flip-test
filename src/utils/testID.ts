import { Platform } from 'react-native'

export default function testID(id: string) {
  return Platform.OS === 'android'
    ? { accessible: true, accessibilityLabel: id }
    : { testID: id }
}
