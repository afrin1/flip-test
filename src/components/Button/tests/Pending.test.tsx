import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import testID from '../../../utils/testID'
import Pending from '../Pending'

describe('Pending Button', () => {
  it('should match snapshot', () => {
    expect(render(
      <Pending
        title={'title'}
        onPress={jest.fn()} />)
    ).toMatchSnapshot()
  })

  it('should display the provided text', () => {
    const { getByText } = render(<Pending
      title={'title'}
      onPress={jest.fn()} />)
    expect(getByText('title')).not.toBeNull()
  })

  it('should invoke the provided callback when button is pressed', () => {
    const onPressMock = jest.fn()
    const { getByTestId } = render(
      <Pending
        {...testID('pending-button')}
        title={'title'}
        onPress={onPressMock} />
    )

    fireEvent.press(getByTestId('pending-button'))
    expect(onPressMock).toBeCalled()
  })
})