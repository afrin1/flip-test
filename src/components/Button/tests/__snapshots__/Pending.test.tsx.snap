// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pending Button should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="pending-button"
>
  <Text
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#FF6347",
        "borderRadius": 10,
        "borderWidth": 2,
        "color": "#000000",
        "fontSize": 14,
        "fontWeight": "bold",
        "margin": 5,
        "padding": 10,
        "textAlignVertical": "center",
      }
    }
  >
    title
  </Text>
</View>
`;
